### PR TITLE
Changes for reader and config files for cosmoDC2 base and test catalogs

### PR DESCRIPTION
**Pull request by evevkovacs**
New config files and reader updates for baseDC2 and cosmoDC2

### CODE REVIEW

File: GCRCatalogs/catalog_configs/baseDC2_v0.4.5.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+subclass_name: cosmodc2.UMGalaxyCatalog
+
+catalog_root_dir: /global/projecta/projectdirs/lsst/groups/CS/cosmoDC2/baseDC2_min_9.8_centrals_v0.4.5
+catalog_path_template: baseDC2_z_{}_{}_cutout_{}.hdf5
+
+cosmology:
+    H0: 71.0
+    Om0: 0.2648
+    Ob0: 0.0448
+
```
Comment (yymao) [inline GCRCatalogs/catalog_configs/baseDC2_v0.4.5.yaml]: add `sigma8` and `n_s`

File: GCRCatalogs/catalog_configs/baseDC2_v0.4.5_test.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+subclass_name: cosmodc2.UMGalaxyCatalog
+
+catalog_root_dir: /global/projecta/projectdirs/lsst/groups/CS/cosmoDC2/baseDC2_min_9.8_centrals_v0.4.5
+catalog_path_template: baseDC2_z_{}_{}_cutout_{}.hdf5
+
+cosmology:
+    H0: 71.0
+    Om0: 0.2648
+    Ob0: 0.0448
+
```
Comment (yymao) [inline GCRCatalogs/catalog_configs/baseDC2_v0.4.5_test.yaml]: add `sigma8` and `n_s`

File: GCRCatalogs/cosmodc2.py [hunk 3]
```diff
@@ -163,7 +166,7 @@ def _get_healpix_file_list(self, assert_files_complete=True):
         possible_healpix_pixel_files = [self._catalog_path_template.format(z, z+1, h) for z in range(self.zrange_lo, self.zrange_hi) for h in self.healpix_pixels]
         healpix_pixel_files = [f for f in possible_healpix_pixel_files if os.path.isfile(f)]
         if assert_files_complete and len(healpix_pixel_files) != len(possible_healpix_pixel_files):
-            raise ValueError('Missing some catalog files')
+            raise ValueError('Missing some catalog files: {}'.format(', '.join([f for f in possible_healpix_pixel_files if not f in healpix_pixel_files])))
```
Comment (yymao) [inline GCRCatalogs/cosmodc2.py]: (optional) use `if f not in` (which is closer to natural language)

File: GCRCatalogs/cosmodc2.py [hunk 4]
```diff
@@ -462,12 +477,17 @@ class UMShearCatalog(UMGalaxyCatalog):
...
+            'convergence': 'conv', 
+            'magnification': (
+                _calc_mag,
+                'conv',
+                'shear_1',
+                'shear_2',
+            ),
             'shear_2_treecorr': 'shear_2',
             'shear_2_phosim':   (np.negative, 'shear_2'),
+            'shear_1':   (np.negative, 'shear_1'),
```
Comment (yymao) [inline GCRCatalogs/cosmodc2.py]: This seems to break the backward compatibility for the shear catalog reader? (cc @patricialarsen)

**Comment by evevkovacs** [comment]
OK, all requested changes have been pushed.

**Comment by evevkovacs** [comment]
Also changed the if not...

**Pull request merged**